tests/test_integration.py

```diff
@@ -23,42 +23,29 @@
 
 import pennylane as qml
 from pennylane.numpy import tensor
-from semantic_version import Version
 import pytest
 import qiskit
 import qiskit_aer
 
 from qiskit.providers import QiskitBackendNotFoundError
-from pennylane_qiskit.qiskit_device import QiskitDevice
+from qiskit.providers.basic_provider import BasicProvider
+from pennylane_qiskit.qiskit_device_legacy import QiskitDeviceLegacy
 
 # pylint: disable=protected-access, unused-argument, ungrouped-imports, too-many-arguments, too-few-public-methods
 
-if Version(qiskit.__version__) < Version("1.0.0"):
-    pldevices = [("qiskit.aer", qiskit_aer.Aer), ("qiskit.basicaer", qiskit.BasicAer)]
 
-    def check_provider_backend_compatibility(pldevice, backend_name):
-        """check compatibility of provided backend"""
-        dev_name, _ = pldevice
-        if (dev_name == "qiskit.aer" and "aer" not in backend_name) or (
-            dev_name == "qiskit.basicaer" and "aer" in backend_name
-        ):
-            return (False, "Only the AerSimulator is supported on AerDevice")
-        return True, None
-
-else:
-    from qiskit.providers.basic_provider import BasicProvider
```

Comment:
Although unrelated to this PR it is deleted since we don't run below v1.0 anymore.